pkg/webhook/resources/version/validator.go

```diff
@@ -55,5 +60,25 @@ func (v *versionValidator) checkAnnotations(version *v1beta1.Version) error {
 
 func (v *versionValidator) Update(_ *types.Request, _ runtime.Object, newObj runtime.Object) error {
 	newVersion := newObj.(*v1beta1.Version)
-	return v.checkAnnotations(newVersion)
+	return checkVersion(newVersion)
+}
+
+func checkVersion(version *v1beta1.Version) error {
+	if err := checkAnnotations(version); err != nil {
+		return err
+	}
+	if err := checkISOChecksum(version); err != nil {
+		return err
+	}
+	return nil
+}
+
+func checkISOChecksum(version *v1beta1.Version) error {
+	isoChecksum := version.Spec.ISOChecksum
+	// if an isoChecksum is provided, it must be in the SHA-512 format
+	// since Longhorn backing images only accept hashes in that format
+	if isoChecksum == "" || SHA512Pattern.MatchString(isoChecksum) {
+		return nil
+	}
+	return werror.NewBadRequest(fmt.Sprintf("invalid isoChecksum %s", isoChecksum))
```

Comment:
I would suggest the following:

```suggestion
	if isoChecksum != "" && !SHA512Pattern.MatchString(isoChecksum) {
		return werror.NewBadRequest(fmt.Sprintf("invalid isoChecksum %s", isoChecksum))
	}
	return nil
```

It's more consistent and idiomatic.

Reply:
Sure, that make sense to me.